src/index.ts

```diff
@@ -61,8 +61,12 @@ const plugin: JupyterFrontEndPlugin<void> = {
         `jupyter-gallery API version out of sync, expected ${expectedVersion}, got ${data.apiVersion}`
       );
     }
-
     const title = data.title === 'Gallery' ? trans.__('Gallery') : data.title;
+
+    // hide the widget if no exhibits are configured
+    if (data.hideGalleryWithoutExhibits && !data.exhibitsConfigured) {
+      return;
```

Comment:
Maybe to make debugging easier:

```suggestion
      console.log('Gallery extension will not add any UI elements because no exhibits are configured');
      return;
```